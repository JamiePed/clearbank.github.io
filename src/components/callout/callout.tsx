import React from 'react'

import * as Types from './callout.types'
import * as Styles from './callout.styles'

const Callout: React.FunctionComponent<Types.CalloutProps> = ({
  colour,
  children
}): JSX.Element => (
  <Styles.Container color={colour}>{children}</Styles.Container>
)

export default Callout
