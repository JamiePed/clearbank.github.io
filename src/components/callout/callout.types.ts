export type ColorOptions = 'blue' | 'green' | 'orange'

export interface CalloutProps {
  colour?: ColorOptions
}
