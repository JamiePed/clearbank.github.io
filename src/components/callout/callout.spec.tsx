import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import 'jest-styled-components'

import Component from './callout'

import * as Types from './callout.types'

let component: ReturnType<typeof render> = null
let fragment: DocumentFragment = null
let root: HTMLElement = null

const defaultProps: Types.CalloutProps = {
  colour: 'blue'
}

beforeEach(() => {
  component = render(<Component {...defaultProps} children='children' />)

  fragment = component.asFragment()
  root = component.container.firstChild
})

afterEach(() => {
  fragment = null
  root = null
})

describe('Component', () => {
  test('matches snapshot', () => {
    expect(root).toMatchSnapshot()
  })

  test('has attributes from props', () => {
    expect(root).toHaveAttribute('color')
  })

  test('has no attributes if missing props', () => {
    component.rerender(<Component />)

    expect(root).not.toHaveAttribute('color')
  })

  test('is not empty', () => {
    expect(fragment.textContent.length).toBeTruthy()
  })
})
