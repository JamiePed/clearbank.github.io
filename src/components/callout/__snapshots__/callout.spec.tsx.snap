// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component matches snapshot 1`] = `
.c0 {
  padding: 20px;
  color: #031563;
  background: #E1EFF9;
  border-radius: 4px;
  border-left: 6px solid #031563;
}

.c0 p,
.c0 div {
  font-size: 16px;
  color: #031563;
  font-family: "Source Sans Pro",sans-serif;
}

<aside
  class="c0"
  color="blue"
>
  children
</aside>
`;
